File: share-code/bai-tap-lap-trinh/cpp/80.cpp
#include <iostream>
#include <cctype>
using namespace std;

int main() {
    string input = "ab@#AB69C!}+21";
    string numberString;
    string letterString;
    string specialCharString;

    for (char c : input) {
        if (isdigit(c)) {
            numberString += c;
        } else if (isalpha(c)) {
            letterString += c;
        } else if (ispunct(c)) {
            specialCharString += c;
        }
    }
    cout << "Chuoi so: " << numberString << endl;
    cout << "Chuoi chu: " << letterString << endl;
    cout << "Chuoi dac biet: " << specialCharString << endl;

    return 0;
}

File: share-code/bai-tap-lap-trinh/python/string/80.py
import re

input_str       = "ab@#AB69C!}+21"
digits          = ""
letters         = ""
special_chars   = ""

# Cách 1:
for char in input_str:
    if char.isdigit():
        digits += char
    elif char.isalpha():
        letters += char
    else:
        special_chars += char

# Cách 2:
# digits = ''.join(re.findall(r'\d', input_str))

# letters = ''.join(re.findall(r'[a-zA-Z]', input_str))

# special_chars = ''.join(re.findall(r'[^a-zA-Z0-9]', input_str))

print("Chuỗi số:", digits)
print("Chuỗi chữ:", letters)
print("Chuỗi đặc biệt:", special_chars)